File: frontend/src/services/employeeService.js
import axiosClient from "../api/axiosClient";

export const getEmployees = async (
    page = 0,
    size = 10
) => {

    const response =
        await axiosClient.get(
            `/employees?page=${page}&size=${size}`
        );

    return response.data;
};

export const createEmployee = async (
    employeeData
) => {

    const response =
        await axiosClient.post(
            "/employees",
            employeeData
        );

    return response.data;
};

export const updateEmployee = async (
    id,
    employeeData
) => {

    const response =
        await axiosClient.put(
            `/employees/${id}`,
            employeeData
        );

    return response.data;
};

export const deactivateEmployee =
    async (id) => {

        const response =
            await axiosClient.patch(
                `/employees/${id}/deactivate`
            );

        return response.data;
    };

export const activateEmployee =
    async (id) => {

        const response =
            await axiosClient.patch(
                `/employees/${id}/activate`
            );

        return response.data;
    };

File: backend/nextalx-api/src/main/java/com/nextalx/service/EmployeeService.java
package com.nextalx.service;

import com.nextalx.dto.request.CreateEmployeeRequest;
import com.nextalx.dto.request.UpdateEmployeeRequest;
import com.nextalx.dto.response.EmployeeResponse;
import org.springframework.data.domain.Page;

public interface EmployeeService {

    EmployeeResponse createEmployee(
            CreateEmployeeRequest request
    );

    EmployeeResponse updateEmployee(
            Long id,
            UpdateEmployeeRequest request
    );

    EmployeeResponse deactivateEmployee(
            Long id
    );

    EmployeeResponse activateEmployee(
            Long id
    );

    Page<EmployeeResponse> getAllEmployees(
            int page,
            int size
    );
}